java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -0,0 +1,251 @@
+
+import com.google.common.collect.Maps;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.InstanceOfTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.ParenthesizedTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeCastTree;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import javax.annotation.CheckForNull;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConstraintManager {
+
+  private final Map<Tree, SymbolicValue> map = new HashMap<>();
+  /**
+   * Map to handle lookup of fields.
+   * */
+  private final Map<Symbol, SymbolicValue> symbolMap = new HashMap<>();
+
+
+
+  public SymbolicValue createSymbolicValue(Tree syntaxNode) {
+    SymbolicValue result = map.get(syntaxNode);
+    if (result == null && syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+      result = symbolMap.get(((IdentifierTree) syntaxNode).symbol());
+    } else if (result == null && syntaxNode.is(Tree.Kind.VARIABLE)) {
+      result = symbolMap.get(((VariableTree) syntaxNode).symbol());
+    }
+    if (result == null) {
+      result = new SymbolicValue.ObjectSymbolicValue(map.size()+ProgramState.EMPTY_STATE.constraints.size());
+      map.put(syntaxNode, result);
+      if(syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+        symbolMap.put(((IdentifierTree) syntaxNode).symbol(), result);
+      } else if(syntaxNode.is(Tree.Kind.VARIABLE)) {
+        symbolMap.put(((VariableTree) syntaxNode).symbol(), result);
+      }
+    }
+    return result;
+  }
+
+
+  public SymbolicValue eval(ProgramState programState, Tree syntaxNode) {
+    syntaxNode = skipTrivial(syntaxNode);
+    switch (syntaxNode.kind()) {
+      case NULL_LITERAL: {
+        return SymbolicValue.NULL_LITERAL;
+      }
+      case BOOLEAN_LITERAL: {
+        boolean value = Boolean.parseBoolean(((LiteralTree) syntaxNode).value());
+        if(value) {
+          return SymbolicValue.TRUE_LITERAL;
+        }
+        return SymbolicValue.FALSE_LITERAL;
+      }
+      case VARIABLE: {
+        Symbol symbol = ((VariableTree) syntaxNode).symbol();
+        SymbolicValue result = programState.values.get(symbol);
+        if (result != null) {
+          // symbolic value associated with local variable
+          return result;
+        }
+        break;
+      }
+      case IDENTIFIER: {
+        Symbol symbol = ((IdentifierTree) syntaxNode).symbol();
+        SymbolicValue result = programState.values.get(symbol);
+        if (result != null) {
+          // symbolic value associated with local variable
+          return result;
+        }
+        break;
+      }
+      case LOGICAL_COMPLEMENT : {
+        UnaryExpressionTree unaryExpressionTree = (UnaryExpressionTree) syntaxNode;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 11. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)